contrib/masterfiles-stage/README.org

```diff
@@ -64,42 +69,76 @@ sure that `masterfiles-stage.sh` is executable.
 
 The `masterfiles-stage.sh` script takes options in order to specify the
 directory where the masterfiles should be deployed after staging, as well as
-the file that contains the paramaters that should be used in order to interface
+the file that contains the parameters that should be used in order to interface
 with the given upstream source implementation (git url, branch etc ..)
+Run masterfiles-stage.sh --help to see exact option flags.
 
-- `MASTERDIR` is the fully qualified path to the location that you
-  want masterfiles to be deployed to after successful staging.
-
-- `PARAMS` is the fully qualified path to the location of the
+- `PARAMS` is the absolute path to the location of the
   parameter set that describes information needed by the selected
-  staging method (currently reffered to as `VCS_TYPE`). The format of
+  staging method (currently referred to as `VCS_TYPE`). The format of
   the file is a simple KEY="VALUE". It is sourced by the
   `masterfiles-stage.sh` script. Each staging method or `VCS_TYPE`
-  defines may define its own parameters. It is reccomended that the
+  defined may define its own parameters. It is recommended that the
   parameters defined by the `VCS_TYPE` be prefixed with the `VCS_TYPE`
   so as not to conflict with other staging methods.
 
   Example PARAMS files can be found [[file:example_params/][here]].
 
+- `MASTERDIR` is the abolute path to the location that you
+  want masterfiles to be deployed to after successful staging.
+  The default is `/var/cfengine/masterfiles` but please note that
+  you can override this value either with an option on the command line,
+  or by specifying the value of `MASTERDIR` inside of your `PARAMS` file.
+  (The value specified in `PARAMS`, if there is one, will take precedence
+  over an option specified on the command line.)
+
+
 *** Reserved Parameters
 The following parameters are in use, be sure to avoid their collision
 when developing new staging methods.
 
 - VCS_TYPE - The staging method to use
-  - Currently supported: (GIT|GIT_TAG_OR_COMMIT|SVN)
+  - Currently supported: (GIT|GIT_POLICY_CHANNELS|SVN)
 
 - GIT_URL - The URL to the git repository
   - Example values:
     - https://gitlab.com/nickanderson/masterfiles_demo_3.7.git
     - git@gitlab.com:nickanderson/masterfiles_demo_3.7.git
 
-- GIT_BRANCH - The git branch to checkout
-  - master
-
-- GIT_TAG_OR_COMMIT
+- GIT_BRANCH - The git refspec to checkout.
+  It can be a branch name, a tag name, a commit hash or a partial hash.
   - Example values:
+    - master
+    - my_tag
     - 88335d36b48c8808b12b48667a463182dc8d0338
-    - mytag
+    - cb375d0f
+
+- dir_to_hold_mirror - The directory where the git mirror repo will
+  be created for GIT_POLICY_CHANNELS.  (With VCS_TYPE=GIT, the repo
+  is placed alongside of ROOT.)
+  - Example value:
+    - /opt/cfengine
```

Comment:
@nickanderson, maybe we could improve the name, and use it for all VCS_TYPEs?  Since this is really what ROOT was used for, but ROOT is something we should leave alone since Design Center uses it...

Maybe call it "Staging Dir" with description "The location where persistent files (such as a git repository directory, or an svn checkout) may be created and accessed by the staging script."